cmake-tests/test-install.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+set -e
+
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
+  DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+  SOURCE="$(readlink "$SOURCE")"
+  [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+done
+SOURCE_DIR="$( cd -P "$( dirname "$SOURCE" )" && cd .. && pwd )"
+
+TMP_DIR=$(mktemp -d)
+
+INSTALL_DIR=$TMP_DIR/hcc
+
+mkdir $TMP_DIR/hcc-build && cd $TMP_DIR/hcc-build
+
+cmake $SOURCE_DIR -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR $@
+make -j`nproc` world
+make -j`nproc`
+make install
+
```

Comment:
Execution of this test should be decoupled from the compilation/installation of hcc.  This test should be able to run in an environment where an hcc compiler has already been installed to a known location without having access to hcc build tree.
The most common use case would be a project using the hcc compiler that has been installed to /opt/rocm/ (or any other known alternative location).  The project needs to be installed to a different directory other than /opt/rocm.  We want a standalone test case that has a find_package(hcc), is able to find the hcc-config from the existing hcc installation and then gets compiled successfully.